File: app/src/main/java/ru/teamdroid/recipecraft/concept/util/schedulers/RunOn.java
package ru.teamdroid.recipecraft.concept.util.schedulers;

import java.lang.annotation.Documented;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import javax.inject.Qualifier;

/**
 * Qualifier to define Scheduler type (io, computation, or ui main thread).
 */

@Qualifier
@Documented
@Retention(RetentionPolicy.RUNTIME)
public @interface RunOn {
  SchedulerType value() default SchedulerType.IO;
}

File: app/src/main/java/ru/teamdroid/recipecraft/util/schedulers/RunOn.kt
package ru.teamdroid.recipecraft.util.schedulers

import javax.inject.Qualifier

/**
 * Qualifier to define Scheduler type (io, computation, or ui main thread).
 */

@Qualifier
@MustBeDocumented
@Retention(AnnotationRetention.RUNTIME)
annotation class RunOn(val value: SchedulerType = SchedulerType.IO)